include/Token.h

```diff
@@ -165,6 +165,8 @@ class Token {
     /// Determines whether the token has the given trivia.
     bool hasTrivia(TriviaKind triviaKind) const;
 
+    bool isKeywordOrIdentifier() const;
```

Comment:
Just a stylistic thing: we can make this a free function (see LexerFacts.cpp) instead of a member?